src/components/EmailEditor.vue

```diff
@@ -6,28 +6,35 @@
   ></div>
 </template>
 
-<script>
-import { loadScript } from './loadScript';
+<script lang="ts">
+import { defineComponent, PropType } from 'vue';
 import pkg from '../../package.json';
-
+import { loadScript } from './loadScript';
+import { EmailEditorProps } from './types';
 let lastEditorId = 0;
 
-export default {
+export default defineComponent({
   name: 'EmailEditor',
   props: {
-    editorId: String,
-    options: Object,
-    projectId: Number,
-    tools: Object,
-    appearance: Object,
-    locale: String,
+    appearance: Object as PropType<EmailEditorProps['appearance']>,
```

Comment:
Do we have a reason to add the types as assertions?

Reply:
I have updated syntax but we need the assertions so typescript can pick up on the types and give autocomplete suggestions, since our types are `types` only and not values i don't see another way to